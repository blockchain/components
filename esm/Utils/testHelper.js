import React from 'react';
import { shallow, mount } from 'enzyme';
import { ThemeProvider } from 'styled-components';
import theme from '../Tools/theme';

function wrapWithTheme(fn, children, options) {
  var wrapper = fn(React.createElement(ThemeProvider, {
    theme: theme
  }, children), options);
  return wrapper[fn.name]({
    context: wrapper.instance().getChildContext()
  });
}

export function shallowWithTheme() {
  return wrapWithTheme.apply(void 0, [shallow].concat(Array.prototype.slice.call(arguments)));
}
export function mountWithTheme() {
  return wrapWithTheme.apply(void 0, [mount].concat(Array.prototype.slice.call(arguments)));
}