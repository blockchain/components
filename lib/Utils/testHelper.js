"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.shallowWithTheme = shallowWithTheme;
exports.mountWithTheme = mountWithTheme;

var _react = _interopRequireDefault(require("react"));

var _enzyme = require("enzyme");

var _styledComponents = require("styled-components");

var _theme = _interopRequireDefault(require("../Tools/theme"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function wrapWithTheme(fn, children, options) {
  var wrapper = fn(_react.default.createElement(_styledComponents.ThemeProvider, {
    theme: _theme.default
  }, children), options);
  return wrapper[fn.name]({
    context: wrapper.instance().getChildContext()
  });
}

function shallowWithTheme() {
  return wrapWithTheme.apply(void 0, [_enzyme.shallow].concat(Array.prototype.slice.call(arguments)));
}

function mountWithTheme() {
  return wrapWithTheme.apply(void 0, [_enzyme.mount].concat(Array.prototype.slice.call(arguments)));
}