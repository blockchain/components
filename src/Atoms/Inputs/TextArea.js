// @flow strict
import React from 'react'
import PropTypes from 'prop-types'
import styled, { type ReactComponentFunctional } from 'styled-components'

import injectE2E, { type E2ePropType } from '../../Tools/injectE2E'
import { prop, theme } from '../../Tools/interpolation'

export type PropsType = {
  +disabled?: boolean,
  +e2e?: string,
  +height?: string,
  +input?: {},
  +name?: string,
  +value?: string,
  +width?: string,
}

type WPropsType = { +name?: string } & E2ePropType

const Wrapper: ReactComponentFunctional<WPropsType> = styled.textarea.attrs({
  'data-e2e': injectE2E,
  name: (props: PropsType) => props.name,
})`
  display: block;
  width: ${prop<PropsType>('width')};
  height: ${prop<PropsType>('height')};
  padding: 0.7rem 1.4rem;
  box-sizing: border-box;
  font-family: ${theme('fontPrimary')};
  font-size: 1rem;
  color: ${theme('inputColor')};
  border-width: 1px;
  border-color: ${theme('inputBorderColor')};
  border-style: solid;
  border-radius: ${theme('inputBorderRadius')};
  background-color: ${theme('inputBackgroundColor')};
  background-image: none;
  outline-width: 0;
  user-select: text;
  resize: none;

  &:disabled {
    color: ${theme('inputColor')};
    background: ${theme('inputBackgroundColor')};
    cursor: not-allowed;
  }

  &::placeholder {
    color: ${theme('inputPlaceholderColor')};
    font-size: 0.875rem;
  }
`

const TextArea = ({ input, ...rest }: PropsType) => {
  return <Wrapper {...input} {...rest} />
}

TextArea.propTypes = {
  disabled: PropTypes.bool,
  e2e: PropTypes.string,
  height: PropTypes.string,
  input: PropTypes.object,
  name: PropTypes.string,
  value: PropTypes.string,
  width: PropTypes.string,
}

TextArea.defaultProps = {
  disabled: false,
  e2e: 'defaultTextArea',
  height: '100px',
  width: '100%',
}

export default TextArea
