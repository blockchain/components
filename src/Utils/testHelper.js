import React from 'react'
import { shallow, mount } from 'enzyme'
import { ThemeProvider } from 'styled-components'
import theme from '../Tools/theme'

function wrapWithTheme(fn, children, options) {
  const wrapper = fn(
    <ThemeProvider theme={theme}>{children}</ThemeProvider>,
    options,
  )

  return wrapper[fn.name]({
    context: wrapper.instance().getChildContext(),
  })
}

export function shallowWithTheme() {
  return wrapWithTheme(shallow, ...arguments)
}

export function mountWithTheme() {
  return wrapWithTheme(mount, ...arguments)
}
