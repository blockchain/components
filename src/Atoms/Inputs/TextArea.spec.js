import React from 'react'
import { mountWithTheme } from '../../Utils/testHelper'

import TextArea from './TextArea'

describe('TextArea', () => {
  it('should render correctly', () => {
    const tree = mountWithTheme(<TextArea />)
    expect(tree.find(TextArea)).toMatchSnapshot()
  })

  it('should render correctly with disabled', () => {
    const tree = mountWithTheme(<TextArea disabled />)
    expect(tree.find(TextArea)).toMatchSnapshot()
  })

  it('should render correctly with error="error"', () => {
    const tree = mountWithTheme(<TextArea error="error" />)

    expect(tree.find(TextArea)).toMatchSnapshot()
  })

  it('should render correctly with label="label"', () => {
    const tree = mountWithTheme(<TextArea label="label" />)

    expect(tree.find(TextArea)).toMatchSnapshot()
  })

  it('should render correctly with label="label" and error="error"', () => {
    const tree = mountWithTheme(<TextArea error="error" label="label" />)

    expect(tree.find(TextArea)).toMatchSnapshot()
  })

  it('should render correctly with e2e', () => {
    const tree = mountWithTheme(<TextArea e2e="E2ELabel" />)
    expect(tree.find(TextArea)).toMatchSnapshot()
  })

  it('should render correctly with height', () => {
    const tree = mountWithTheme(<TextArea height="200px" />)
    expect(tree.find(TextArea)).toMatchSnapshot()
  })

  it('should render correctly with name', () => {
    const tree = mountWithTheme(<TextArea name="mytextarea" />)
    expect(tree.find(TextArea)).toMatchSnapshot()
  })

  it('should render correctly with value', () => {
    const tree = mountWithTheme(
      <TextArea onChange={(n) => n} value="Hello dear friend" />,
    )
    expect(tree.find(TextArea)).toMatchSnapshot()
  })

  it('should render correctly with width', () => {
    const tree = mountWithTheme(<TextArea width="200px" />)
    expect(tree.find(TextArea)).toMatchSnapshot()
  })
})
