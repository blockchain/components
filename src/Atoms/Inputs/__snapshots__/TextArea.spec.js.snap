// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TextArea should render correctly 1`] = `
.c0 {
  display: block;
  width: 100%;
  height: 100px;
  padding: 0.7rem 1.4rem;
  box-sizing: border-box;
  font-family: 'IBM Plex Sans','Helvetica Neue',Arial,sans-serif;
  font-size: 1rem;
  color: #545456;
  border-width: 1px;
  border-color: #CCC;
  border-style: solid;
  border-radius: 0.125rem;
  background-color: #FFF;
  background-image: none;
  outline-width: 0;
  -webkit-user-select: text;
  -moz-user-select: text;
  -ms-user-select: text;
  user-select: text;
  resize: none;
}

.c0:disabled {
  color: #545456;
  background: #FFF;
  cursor: not-allowed;
}

.c0::-webkit-input-placeholder {
  color: #999B9E;
  font-size: 0.875rem;
}

.c0::-moz-placeholder {
  color: #999B9E;
  font-size: 0.875rem;
}

.c0:-ms-input-placeholder {
  color: #999B9E;
  font-size: 0.875rem;
}

.c0::placeholder {
  color: #999B9E;
  font-size: 0.875rem;
}

<TextArea
  disabled={false}
  e2e="defaultTextArea"
  height="100px"
  width="100%"
>
  <styled.textarea
    disabled={false}
    e2e="defaultTextArea"
    height="100px"
    width="100%"
  >
    <textarea
      className="c0"
      data-e2e="defaultTextArea"
      disabled={false}
      height="100px"
      width="100%"
    />
  </styled.textarea>
</TextArea>
`;

exports[`TextArea should render correctly with disabled 1`] = `
.c0 {
  display: block;
  width: 100%;
  height: 100px;
  padding: 0.7rem 1.4rem;
  box-sizing: border-box;
  font-family: 'IBM Plex Sans','Helvetica Neue',Arial,sans-serif;
  font-size: 1rem;
  color: #545456;
  border-width: 1px;
  border-color: #CCC;
  border-style: solid;
  border-radius: 0.125rem;
  background-color: #FFF;
  background-image: none;
  outline-width: 0;
  -webkit-user-select: text;
  -moz-user-select: text;
  -ms-user-select: text;
  user-select: text;
  resize: none;
}

.c0:disabled {
  color: #545456;
  background: #FFF;
  cursor: not-allowed;
}

.c0::-webkit-input-placeholder {
  color: #999B9E;
  font-size: 0.875rem;
}

.c0::-moz-placeholder {
  color: #999B9E;
  font-size: 0.875rem;
}

.c0:-ms-input-placeholder {
  color: #999B9E;
  font-size: 0.875rem;
}

.c0::placeholder {
  color: #999B9E;
  font-size: 0.875rem;
}

<TextArea
  disabled={true}
  e2e="defaultTextArea"
  height="100px"
  width="100%"
>
  <styled.textarea
    disabled={true}
    e2e="defaultTextArea"
    height="100px"
    width="100%"
  >
    <textarea
      className="c0"
      data-e2e="defaultTextArea"
      disabled={true}
      height="100px"
      width="100%"
    />
  </styled.textarea>
</TextArea>
`;

exports[`TextArea should render correctly with e2e 1`] = `
.c0 {
  display: block;
  width: 100%;
  height: 100px;
  padding: 0.7rem 1.4rem;
  box-sizing: border-box;
  font-family: 'IBM Plex Sans','Helvetica Neue',Arial,sans-serif;
  font-size: 1rem;
  color: #545456;
  border-width: 1px;
  border-color: #CCC;
  border-style: solid;
  border-radius: 0.125rem;
  background-color: #FFF;
  background-image: none;
  outline-width: 0;
  -webkit-user-select: text;
  -moz-user-select: text;
  -ms-user-select: text;
  user-select: text;
  resize: none;
}

.c0:disabled {
  color: #545456;
  background: #FFF;
  cursor: not-allowed;
}

.c0::-webkit-input-placeholder {
  color: #999B9E;
  font-size: 0.875rem;
}

.c0::-moz-placeholder {
  color: #999B9E;
  font-size: 0.875rem;
}

.c0:-ms-input-placeholder {
  color: #999B9E;
  font-size: 0.875rem;
}

.c0::placeholder {
  color: #999B9E;
  font-size: 0.875rem;
}

<TextArea
  disabled={false}
  e2e="E2ELabel"
  height="100px"
  width="100%"
>
  <styled.textarea
    disabled={false}
    e2e="E2ELabel"
    height="100px"
    width="100%"
  >
    <textarea
      className="c0"
      data-e2e="E2ELabel"
      disabled={false}
      height="100px"
      width="100%"
    />
  </styled.textarea>
</TextArea>
`;

exports[`TextArea should render correctly with height 1`] = `
.c0 {
  display: block;
  width: 100%;
  height: 200px;
  padding: 0.7rem 1.4rem;
  box-sizing: border-box;
  font-family: 'IBM Plex Sans','Helvetica Neue',Arial,sans-serif;
  font-size: 1rem;
  color: #545456;
  border-width: 1px;
  border-color: #CCC;
  border-style: solid;
  border-radius: 0.125rem;
  background-color: #FFF;
  background-image: none;
  outline-width: 0;
  -webkit-user-select: text;
  -moz-user-select: text;
  -ms-user-select: text;
  user-select: text;
  resize: none;
}

.c0:disabled {
  color: #545456;
  background: #FFF;
  cursor: not-allowed;
}

.c0::-webkit-input-placeholder {
  color: #999B9E;
  font-size: 0.875rem;
}

.c0::-moz-placeholder {
  color: #999B9E;
  font-size: 0.875rem;
}

.c0:-ms-input-placeholder {
  color: #999B9E;
  font-size: 0.875rem;
}

.c0::placeholder {
  color: #999B9E;
  font-size: 0.875rem;
}

<TextArea
  disabled={false}
  e2e="defaultTextArea"
  height="200px"
  width="100%"
>
  <styled.textarea
    disabled={false}
    e2e="defaultTextArea"
    height="200px"
    width="100%"
  >
    <textarea
      className="c0"
      data-e2e="defaultTextArea"
      disabled={false}
      height="200px"
      width="100%"
    />
  </styled.textarea>
</TextArea>
`;

exports[`TextArea should render correctly with name 1`] = `
.c0 {
  display: block;
  width: 100%;
  height: 100px;
  padding: 0.7rem 1.4rem;
  box-sizing: border-box;
  font-family: 'IBM Plex Sans','Helvetica Neue',Arial,sans-serif;
  font-size: 1rem;
  color: #545456;
  border-width: 1px;
  border-color: #CCC;
  border-style: solid;
  border-radius: 0.125rem;
  background-color: #FFF;
  background-image: none;
  outline-width: 0;
  -webkit-user-select: text;
  -moz-user-select: text;
  -ms-user-select: text;
  user-select: text;
  resize: none;
}

.c0:disabled {
  color: #545456;
  background: #FFF;
  cursor: not-allowed;
}

.c0::-webkit-input-placeholder {
  color: #999B9E;
  font-size: 0.875rem;
}

.c0::-moz-placeholder {
  color: #999B9E;
  font-size: 0.875rem;
}

.c0:-ms-input-placeholder {
  color: #999B9E;
  font-size: 0.875rem;
}

.c0::placeholder {
  color: #999B9E;
  font-size: 0.875rem;
}

<TextArea
  disabled={false}
  e2e="defaultTextArea"
  height="100px"
  name="mytextarea"
  width="100%"
>
  <styled.textarea
    disabled={false}
    e2e="defaultTextArea"
    height="100px"
    name="mytextarea"
    width="100%"
  >
    <textarea
      className="c0"
      data-e2e="defaultTextArea"
      disabled={false}
      height="100px"
      name="mytextarea"
      width="100%"
    />
  </styled.textarea>
</TextArea>
`;

exports[`TextArea should render correctly with value 1`] = `
.c0 {
  display: block;
  width: 100%;
  height: 100px;
  padding: 0.7rem 1.4rem;
  box-sizing: border-box;
  font-family: 'IBM Plex Sans','Helvetica Neue',Arial,sans-serif;
  font-size: 1rem;
  color: #545456;
  border-width: 1px;
  border-color: #CCC;
  border-style: solid;
  border-radius: 0.125rem;
  background-color: #FFF;
  background-image: none;
  outline-width: 0;
  -webkit-user-select: text;
  -moz-user-select: text;
  -ms-user-select: text;
  user-select: text;
  resize: none;
}

.c0:disabled {
  color: #545456;
  background: #FFF;
  cursor: not-allowed;
}

.c0::-webkit-input-placeholder {
  color: #999B9E;
  font-size: 0.875rem;
}

.c0::-moz-placeholder {
  color: #999B9E;
  font-size: 0.875rem;
}

.c0:-ms-input-placeholder {
  color: #999B9E;
  font-size: 0.875rem;
}

.c0::placeholder {
  color: #999B9E;
  font-size: 0.875rem;
}

<TextArea
  disabled={false}
  e2e="defaultTextArea"
  height="100px"
  value="Hello dear friend"
  width="100%"
>
  <styled.textarea
    disabled={false}
    e2e="defaultTextArea"
    height="100px"
    value="Hello dear friend"
    width="100%"
  >
    <textarea
      className="c0"
      data-e2e="defaultTextArea"
      disabled={false}
      height="100px"
      value="Hello dear friend"
      width="100%"
    />
  </styled.textarea>
</TextArea>
`;

exports[`TextArea should render correctly with width 1`] = `
.c0 {
  display: block;
  width: 200px;
  height: 100px;
  padding: 0.7rem 1.4rem;
  box-sizing: border-box;
  font-family: 'IBM Plex Sans','Helvetica Neue',Arial,sans-serif;
  font-size: 1rem;
  color: #545456;
  border-width: 1px;
  border-color: #CCC;
  border-style: solid;
  border-radius: 0.125rem;
  background-color: #FFF;
  background-image: none;
  outline-width: 0;
  -webkit-user-select: text;
  -moz-user-select: text;
  -ms-user-select: text;
  user-select: text;
  resize: none;
}

.c0:disabled {
  color: #545456;
  background: #FFF;
  cursor: not-allowed;
}

.c0::-webkit-input-placeholder {
  color: #999B9E;
  font-size: 0.875rem;
}

.c0::-moz-placeholder {
  color: #999B9E;
  font-size: 0.875rem;
}

.c0:-ms-input-placeholder {
  color: #999B9E;
  font-size: 0.875rem;
}

.c0::placeholder {
  color: #999B9E;
  font-size: 0.875rem;
}

<TextArea
  disabled={false}
  e2e="defaultTextArea"
  height="100px"
  width="200px"
>
  <styled.textarea
    disabled={false}
    e2e="defaultTextArea"
    height="100px"
    width="200px"
  >
    <textarea
      className="c0"
      data-e2e="defaultTextArea"
      disabled={false}
      height="100px"
      width="200px"
    />
  </styled.textarea>
</TextArea>
`;
